internal/server/evaluation/evaluation.go

```diff
@@ -58,6 +60,74 @@ func (e *EvaluateServer) Variant(ctx context.Context, v *rpcEvaluation.Evaluatio
 	span := trace.SpanFromContext(ctx)
 	span.SetAttributes(spanAttrs...)
 
-	e.logger.Debug("evaluate", zap.Stringer("response", resp))
+	s.logger.Debug("evaluate", zap.Stringer("response", resp))
 	return ver, nil
 }
+
+// Boolean evaluates a request for a boolean flag and entity.
+func (s *Server) Boolean(ctx context.Context, r *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.BooleanEvaluationResponse, error) {
+	rollouts, err := s.store.GetEvaluationRollouts(ctx, r.FlagKey, r.NamespaceKey)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &rpcEvaluation.BooleanEvaluationResponse{}
+
+	var lastRank int32
+
+	for _, rollout := range rollouts {
+		if rollout.Rank < lastRank {
+			resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+			return resp, errors.ErrInvalidf("rollout rank: %d detected out of order", rollout.Rank)
+		}
+
+		lastRank = rollout.Rank
+
+		if rollout.Percentage != nil {
+			// consistent hashing based on the entity id and flag key.
+			hash := crc32.ChecksumIEEE([]byte(r.EntityId + r.FlagKey))
+
+			normalizedValue := float32(int(hash) % 100)
+
+			// if this case does not hold, fall through to the next rollout.
+			if normalizedValue < rollout.Percentage.Percentage {
+				resp.Value = rollout.Percentage.Value
+				resp.Reason = rpcEvaluation.EvaluationReason_MATCH_EVALUATION_REASON
+				s.logger.Debug("percentage based matched", zap.Int("rank", int(rollout.Rank)), zap.String("rollout_type", "percentage"))
+
+				return resp, nil
+			}
+		} else if rollout.Segment != nil {
+			matched, err := doConstraintsMatch(s.logger, r.Context, rollout.Segment.Constraints, rollout.Segment.SegmentMatchType)
+			if err != nil {
+				resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+				return resp, err
+			}
+
+			// if we don't match the segment, fall through to the next rollout.
+			if !matched {
+				continue
+			}
+
+			resp.Value = rollout.Segment.Value
+			resp.Reason = rpcEvaluation.EvaluationReason_MATCH_EVALUATION_REASON
+
+			s.logger.Debug("segment based matched", zap.Int("rank", int(rollout.Rank)), zap.String("segment", rollout.Segment.SegmentKey))
+
+			return resp, nil
+		}
+	}
+
+	f, err := s.store.GetFlag(ctx, r.NamespaceKey, r.FlagKey)
+	if err != nil {
+		resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
```

Comment:
what happens if `flag not found` here? shouldn't we return that reason?

Reply:
@markphelps Yeah agreed. We should check to see if it is not found and the use that as a reason.